### PR TITLE
Mocking template provider

### PR DESCRIPTION
**Pull request by lorbax**

### CODE REVIEW

File: utils/message-generator/debug [hunk 1]
```diff
@@ -0,0 +1,55 @@
+  0,
```
Comment (darricksee) [inline utils/message-generator/debug:1]: Where is this new file used?

File: protocols/v2/subprotocols/common-messages/Cargo.toml [hunk 2]
```diff
@@ -14,7 +14,8 @@ binary_sv2 = {version = "0.1.*", path = "../../../../protocols/v2/binary-sv2/bin
 const_sv2 = {version = "0.1.*", path = "../../../../protocols/v2/const-sv2"}
 quickcheck = { version = "1.0.3", optional=true }
 quickcheck_macros = { version = "1", optional=true }
+serde_repr = {version= "0.1.10", optional=true}
```
Comment (darricksee) [inline protocols/v2/subprotocols/common-messages/Cargo.toml:17]: Can you explain why we need this?

Comment (lorbax) [reply protocols/v2/subprotocols/common-messages/Cargo.toml:17]: I had problem in parsing type Protocol, so I used serve_repr
https://github.com/stratum-mining/stratum/blob/main/protocols/v2/subprotocols/common-messages/src/setup_connection.rs#L287

Comment (lorbax) [reply protocols/v2/subprotocols/common-messages/Cargo.toml:17]: https://github.com/stratum-mining/stratum/pull/383/commits/db1936bc365dd6be0ab13fc13e52de31763eea2c

File: utils/message-generator/debug (hunk 1 above)
Comment (lorbax) [reply utils/message-generator/debug:1]: I needed that file to fix a bug of the MG, I don't know how it is included in the commit. I'll remove it.

**Pull request merged**